src/contact.rs

```diff
@@ -34,34 +30,60 @@ use crate::{chat, stock_str};
 ///
 /// Some contact IDs are reserved to identify special contacts.  This
 /// type can represent both the special as well as normal contacts.
-#[derive(
-    Debug, Copy, Clone, Default, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize,
-)]
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct ContactId(u32);
 
 impl ContactId {
+    pub const UNDEFINED: ContactId = ContactId::new(0);
+    /// The owner of the account.
+    ///
+    /// The email-address is set by `dc_set_config` using "addr".
+    pub const SELF: ContactId = ContactId::new(1);
+    pub const INFO: ContactId = ContactId::new(2);
+    pub const DEVICE: ContactId = ContactId::new(5);
+    const LAST_SPECIAL: ContactId = ContactId::new(9);
+
+    /// Address to go with [`ContactId::DEVICE`].
+    ///
+    /// This is used by APIs which need to return an email address for this contact.
+    pub const DEVICE_ADDR: &'static str = "device@localhost";
+
     /// Creates a new [`ContactId`].
     pub const fn new(id: u32) -> ContactId {
         ContactId(id)
     }
 
-    /// Bad evil escape hatch, do not use.
+    /// Whether this is a special [`ContactId`].
+    ///
+    /// Some [`ContactId`]s are reserved for special contacts like [`ContactId::SELF`],
+    /// [`ContactId::INFO`] and [`ContactId::DEVICE`].  This function indicates whether this
+    /// [`ContactId`] is any of the reserved special [`ContactId`]s (`true`) or whether it
+    /// is the [`ContactId`] of a real contact (`false`).
+    pub fn is_special(&self) -> bool {
+        self.0 <= Self::LAST_SPECIAL.0
+    }
+
+    /// Numerical representation of the [`ContactId`].
+    ///
+    /// Each contact ID has a unique numerical representation which is used in the database
+    /// (via [`rusqlite::ToSql`]) and also for FFI purposes.  In Rust code you should never
+    /// need to use this directly.
     pub const fn to_u32(&self) -> u32 {
         self.0
     }
 }
 
 impl fmt::Display for ContactId {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        if *self == DC_CONTACT_ID_UNDEFINED {
+        if *self == ContactId::UNDEFINED {
             write!(f, "Contact#Undefined")
-        } else if *self == DC_CONTACT_ID_SELF {
+        } else if *self == ContactId::SELF {
             write!(f, "Contact#Self")
-        } else if *self == DC_CONTACT_ID_INFO {
+        } else if *self == ContactId::INFO {
             write!(f, "Contact#Info")
-        } else if *self == DC_CONTACT_ID_DEVICE {
+        } else if *self == ContactId::DEVICE {
             write!(f, "Contact#Device")
-        } else if *self <= DC_CONTACT_ID_LAST_SPECIAL {
+        } else if self.is_special() {
```

Comment:
in practice, there is no other "special case" left - so we could probably also just do the default (which would be fine if we add another special as well)